deployment/scripts/clusters.backup-multiple-servers.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# time stamp
+printf "\nMultiple backups started at:\n  "
+date
+
+# base setup
+SCRIPT_PATH=$(realpath $0)
+SCRIPT_DIR=$(dirname $SCRIPT_PATH)
+
+# save old CONFIGURATION for later reset
+export SAVE_CONFIGURATION=$CONFIGURATION
+
+# export all variables in "../.env"
+set -a            
+source ${SCRIPT_DIR}/../.env
+set +a
+
+# check BACKUP_CONFIGURATIONS
+if [[ -z ${BACKUP_CONFIGURATIONS} ]]; then
+  #%! echo "You must provide a BACKUP_CONFIGURATIONS via environment variable"
+  printf "!!! You must provide a BACKUP_CONFIGURATIONS via environment variable !!!\n"
+  exit 1
+else
+  # convert configurations to array
+  IFS=' ' read -a CONFIGURATIONS_ARRAY <<< "$BACKUP_CONFIGURATIONS"
+
+  # display the clusters
+  printf "Backup the clusters:\n"
+  for i in "${CONFIGURATIONS_ARRAY[@]}"
+  do
+    echo "  $i"
+  done
+fi
+
+# check BACKUP_SAVED_BACKUPS_NUMBER
+if [[ -z ${BACKUP_SAVED_BACKUPS_NUMBER} ]]; then
+  #%! echo "You must provide a BACKUP_SAVED_BACKUPS_NUMBER via environment variable"
+  printf "!!! You must provide a BACKUP_SAVED_BACKUPS_NUMBER via environment variable !!!\n"
+  exit 1
+else
+  # deleting backups?
+  if (( BACKUP_SAVED_BACKUPS_NUMBER >= 1 )); then
+    printf "Keep the last %d backups for all networks.\n" $BACKUP_SAVED_BACKUPS_NUMBER
+  else
+    echo "!!! ATTENTION: No backups are deleted !!!"
+  fi
+fi
+
+echo "Cancel by ^C. You have 15 seconds"
+# wait for the admin to react
+sleep 15
+
+printf "\n"
+
+for i in "${CONFIGURATIONS_ARRAY[@]}"
+do
+  export CONFIGURATION=$i
+  # individual cluster backup
+  ${SCRIPT_DIR}/cluster.backup.sh
+
+  # deleting backups?
+  if (( BACKUP_SAVED_BACKUPS_NUMBER >= 1 )); then
+    # delete all oldest backups, but leave the last BACKUP_SAVED_BACKUPS_NUMBER
+
+    keep=$BACKUP_SAVED_BACKUPS_NUMBER
+    path="$SCRIPT_DIR/../configurations/$CONFIGURATION/backup/"
+
+    cd $path
+
+    printf "In\n  '$path'\n  remove:\n"
+    while [ `ls -1  |  wc -l` -gt $keep ]; do
+      # TODO: because 'ls' is not always relyable the same on different shells maybe replace 'ls' by 'find . -type d -maxdepth 1'? description: https://unix.stackexchange.com/questions/28939/how-to-delete-the-oldest-directory-in-a-given-directory
+      #         I tested this, but 'find' is crutial to use, because of shell compatibilities
+      oldest=`ls -c1  |  sort -n  |  head -1`
+      printf "  %s\n" $oldest
+      rm -rf $oldest
+    done
+
+    printf "Keep the last %d backups:\n" $BACKUP_SAVED_BACKUPS_NUMBER
+    ls -c1  |  sort -n  |  awk '{print "  " $0}'
+
+    cd $SCRIPT_DIR
+  else
+    echo "!!! ATTENTION: No backups are deleted !!!"
+  fi
```

Comment:
remove else - you have that above

Reply:
Good catch.

But it was intentional and I like to keep this.